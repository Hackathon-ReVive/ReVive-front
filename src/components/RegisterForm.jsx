import { useForm } from "react-hook-form";
import Button from "./Button";
import axios from "axios";
import { useNavigate } from "react-router-dom";

function RegisterForm() {
  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm();
  const navigate = useNavigate();

  const onSubmit = async (data) => {
    try {
      console.log(data);
      const response = await axios.post("AQUI VA LA API", data);
      console.log("Registro Exitoso:", response.data);
      navigate("/login");
    } catch (error) {
      console.error(
        "Error en el Registro:",
        error.response?.data || error.message
      );
    }
  };

  return (
    <div className="max-w-md mx-auto p-6 bg-white shadow-md rounded-md border-stone-200 mt-50">
      <h2 className="text-2xl font-semibold text-center text-gray-700 mb-4">
        Registro de Usuario
      </h2>
      <form onSubmit={handleSubmit(onSubmit)} className="space-y-4">
        <input
          type="text"
          {...register("name", { required: "El nombre es obligatorio" })}
          placeholder="Nombre Completo"
          className="w-full p-2 border border-gray-300 rounded"
        />
        {errors.name && (
          <p className="text-red-500 text-sm">{errors.name.message}</p>
        )}

        <input
          type="email"
          {...register("email", { required: "El correo es obligatorio" })}
          placeholder="Correo Electrónico"
          className="w-full p-2 border border-gray-300 rounded"
        />
        {errors.email && (
          <p className="text-red-500 text-sm">{errors.email.message}</p>
        )}

        <input
          type="password"
          {...register("password", {
            required: "La contraseña es obligatoria",
          })}
          placeholder="Contraseña"
          className="w-full p-2 border border-gray-300 rounded"
        />
        {errors.password && (
          <p className="text-red-500 text-sm">{errors.password.message}</p>
        )}

        <input
          type="password"
          {...register("confirmPassword", {
            required: "Por favor, confirma tu contraseña",
            validate: (value) =>
              value === watch("password") || "Las contraseñas no coinciden",
          })}
          placeholder="Confirmar Contraseña"
          className="w-full p-2 border border-gray-300 rounded"
        />
        {errors.confirmPassword && (
          <p className="text-red-500 text-sm">
            {errors.confirmPassword.message}
          </p>
        )}

        <Button type="submit" text="Registrarse" className="w-full" />
      </form>
    </div>
  );
}

export default RegisterForm;
