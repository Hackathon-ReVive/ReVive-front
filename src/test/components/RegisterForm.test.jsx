import "@testing-library/jest-dom";
import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import RegisterForm from "../../components/RegisterForm";



jest.mock("../../services/api", () => ({
    registerUser: jest.fn(),
}));

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockNavigate,
}));

describe("RegisterForm Component", () => {
    test("renders the form with all input fields", () => {
        render(
            <MemoryRouter>
                <RegisterForm />
            </MemoryRouter>
        );

        expect(screen.getByPlaceholderText("Nombre Completo")).toBeInTheDocument();
        expect(screen.getByPlaceholderText("Correo Electrónico")).toBeInTheDocument();
        expect(screen.getByPlaceholderText("Dirección")).toBeInTheDocument();
        expect(screen.getByPlaceholderText("Código Postal")).toBeInTheDocument();
        expect(screen.getByPlaceholderText("Contraseña")).toBeInTheDocument();
        expect(screen.getByPlaceholderText("Confirmar Contraseña")).toBeInTheDocument();
    });

    test("shows validation errors when submitting empty fields", async () => {
        render(
            <MemoryRouter>
                <RegisterForm />
            </MemoryRouter>
        );

        fireEvent.click(screen.getByRole("button", { name: /Registrarse/i }));

        expect(await screen.findByText(/El nombre es obligatorio/i)).toBeInTheDocument();
        expect(await screen.findByText(/El correo es obligatorio/i)).toBeInTheDocument();
        expect(await screen.findByText(/La dirección es obligatoria/i)).toBeInTheDocument();
        expect(await screen.findByText(/El código postal es obligatorio/i)).toBeInTheDocument();
        expect(await screen.findByText(/La contraseña es obligatoria/i)).toBeInTheDocument();
        expect(await screen.findByText(/Por favor, confirma tu contraseña/i)).toBeInTheDocument();
    });


    test("shows an error when passwords do not match", async () => {
        render(
            <MemoryRouter>
                <RegisterForm />
            </MemoryRouter>
        );

        fireEvent.change(screen.getByPlaceholderText("Contraseña"), { target: { value: "password123" } });
        fireEvent.change(screen.getByPlaceholderText("Confirmar Contraseña"), { target: { value: "differentPassword" } });
        fireEvent.click(screen.getByRole("button", { name: /Registrarse/i }));

        expect(await screen.findByText(/Las contraseñas no coinciden/i)).toBeInTheDocument();
    });

});
