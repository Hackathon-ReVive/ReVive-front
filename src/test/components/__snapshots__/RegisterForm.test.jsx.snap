// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="min-h-screen flex items-center justify-center bg-cover bg-center bg-[url(/assets/bgLoginRegister.png)]"
  >
    <div
      class="max-w-md mx-auto p-8 bg-[#ECFDF5] shadow-md rounded-3xl border border-green-300 backdrop-blur-lg bg-opacity-80"
    >
      <h2
        class="text-3xl font-bold text-center text-[#064E3B] mb-2"
      >
        Registro de Usuario
      </h2>
      <form
        class="space-y-4"
      >
        <label
          class="block text-[#059669] text-sm font-medium"
          for="name"
        >
          Nombre Completo
        </label>
        <input
          class="w-full p-3 border border-[#A7F3D0] rounded-lg"
          id="name"
          name="name"
          placeholder="Nombre Completo"
          type="text"
        />
        <label
          class="block text-[#059669] text-sm font-medium"
          for="email"
        >
          Correo Electrónico
        </label>
        <input
          class="w-full p-3 border border-[#A7F3D0] rounded-lg"
          id="email"
          name="email"
          placeholder="Correo Electrónico"
          type="email"
        />
        <label
          class="block text-[#059669] text-sm font-medium"
          for="address"
        >
          Dirección
        </label>
        <input
          class="w-full p-3 border border-[#A7F3D0] rounded-lg"
          id="address"
          name="address"
          placeholder="Dirección"
          type="text"
        />
        <label
          class="block text-[#059669] text-sm font-medium"
          for="cp"
        >
          Código Postal
        </label>
        <input
          class="w-full p-3 border border-[#A7F3D0] rounded-lg"
          id="cp"
          name="postalCode"
          placeholder="Código Postal"
          type="text"
        />
        <label
          class="block text-[#059669] text-sm font-medium"
          for="password"
        >
          Contraseña
        </label>
        <input
          class="w-full p-3 border border-[#A7F3D0] rounded-lg"
          id="password"
          name="password"
          placeholder="Contraseña"
          type="password"
        />
        <label
          class="block text-[#059669] text-sm font-medium"
          for="confirmPassword"
        >
          Confirmar Contraseña
        </label>
        <input
          class="w-full p-3 border border-[#A7F3D0] rounded-lg"
          id="confirmPassword"
          name="confirmPassword"
          placeholder="Confirmar Contraseña"
          type="password"
        />
        <p
          class="text-xs text-[#059669] text-center mt-4"
        >
          Al Registrarte, aceptas nuestros 
          <a
            class="text-green-700 underline"
            href="#"
          >
            Terminos de servicio
          </a>
           y 
          <a
            class="text-green-700 underline"
            href="#"
          >
            Políticas de Privacidad
          </a>
          .
        </p>
        <button
          class="bg-teal-500 text-white p-2 rounded hover:bg-teal-600 hover:text-white cursor-pointer transition w-full"
          type="submit"
        >
          Registrarse
        </button>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
